compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1589,10 +1647,11 @@ impl<'a> Parser<'a> {
         adt_ty: &str,
         lo: Span,
         vis: Visibility,
+        mut_restriction: Restriction<ast::restriction_kind::Mut>,
```

Comment:
Would an `type MutRestriction = Restriction<ast::restriction_kind::Mut>;`-kind of alias make sense here?